### DISCUSSION TITLE
Destroy parallel route

### DISCUSSION THREAD

**Discussion by stigmat4j** [vercel/next.js#51824] [App Router]
Is there any way to destroy parallel routing without router.back()?

**Comment by akomm** [comment]
I figured out parallel routes, except those pathless that always match, are a huge pile of mess. The documented behavior is either incomplete and relevant pieces are missing or the behavior is not the documented one. The documentation for example claims you can dismiss dialog using either router.back() or Link, but the later isn't true, because it does typically a soft-reload and maintains the modal.

To achieve desired behavior you have a literal clusterfuck of default.tsx catchall (..) and such. A huge huge pain and red alert flags on this feature.

**Reply by akomm** [reply]
My current solution for modal is without using parallel routes at all. Example:

/files:
layout.tsx {p.children} used for modals only
page.tsx simply returns null
/files/upload:
page.tsx the modal

This is more limiting, than parallel routes IF they worked properly and where not such a mess to use. But it works as expected.

Example content layout.tsx
```tsx
export default function FilesLayout() {
  return (
    <div>
     {p.children} {/* <--- the modal is loaded here if you navigate to /files/upload */}
     {/* shared layout */}
     <FilesTable />
     <WhateverElseContentYouWantAlwaysPresentBehindTheModal />
    </div>
  )
}
```

**Reply by akomm** [reply]
Basically when you are /files/upload/page.tsx within is loaded, if not, p.children is null via files/page.tsx returning null

**Reply by akomm** [reply]
There is an alternative:

/users/@modal/create/
- page.tsx -> the CreateUserModal()

/users/@modal/
- default.tsx --> null
- page.tsx  --> null

/users/
- layout.tsx -> renders props.modal & props.children
- page.tsx -> renders UsersPage()
- default.tsx -> export default UsersPage from "./page.tsx"

Now you don't necessary need to `route.back()`. If you navigate to /users, the modal will close, because the `/users/@modal/page.tsx` matches, which returns null. In case of full reload, the `/users/@modal/default.tsx` will render null. You can use both `Link`, but also `router.push`

TBH I don't like that cluttering and page & default spam.
